common/parser.hpp

```diff
@@ -209,9 +209,12 @@ inline void preprocess_data_payload(std::vector<std::string>& params)
         auto data = std::vector<uint8_t>((std::istreambuf_iterator<char>(file)),
             std::istreambuf_iterator<char>());
 
-        std::vector<std::string> modified_params(params.size() - 1 + data.size());
-        std::transform(data.begin(), data.end(), modified_params.begin() + (params.size() - 1), [](uint8_t c) { return utilities::string::hexify(c); });
-        params = modified_params; // substitute original params with raw data
+        size_t unmodified_params = params.size() - 1; // non-negative due to the file parameter
+        std::vector<std::string> modified_params(unmodified_params + data.size());
+        if (unmodified_params)                        // copy leading parameters
+            std::copy(modified_params.begin(), modified_params.begin() + unmodified_params - 1,params.begin());
```

Comment:
`unmodified_params` already does `-1`, why again?

Reply:
When copying parameters, the `container.begin()` already points to the first item, thus the last in range is `.begin()+params-1`

Reply:
end() is never part of the range: it's `[)`. Iterating over `x..x` is empty.

Reply:
correct